addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6695,29 +6695,46 @@ registry.ImageTools = ImageHandlerOption.extend({
      */
```

Comment:
About the history issue
> Not fixed. It seems the issue is related to using 'this.trigger_up("option_update")' to enable the "hover animation" option. I can't make the history take two steps in one step (changing animation mode + triggering to activate hover effects).

Maybe we can forget it for now (so this PR can be merged) :thinking:  But it should still be fixed later: I tested on runbot (so without this PR) and the hover effect history in general has the issue so it is not linked to the previews you did in this PR :+1: 

FYI, you can make multiple steps in one step :wink:  => you need to call `this.wysiwyg.odooEditor.historyPauseSteps()` at the beginning and then `this.wysiwyg.odooEditor.historyUnpauseSteps()` at the end (you may need to register the step manually afterwards with `this.wysiwyg.odooEditor.historyStep()`)
It allows to ignore every `historyStep()` that are called during the time it is paused :wink:

Reply:
Regarding the first history bug: when transitioning from one hover effect to another and then performing "undo," I may have had a solution, but it was quite cumbersome => listen to the `historyUndo` and `historyRedo` events from the options, then refresh the widget at that time. However, it still wasn't perfect because we still needed to figure out how to destroy the widget before undo/redo. Finally, I thought it was becoming overly complicated for a non-critical bug; the user can simply reapply the previous effect via the options.

However, for the other bug, "Drop Text-Image and set the image animation to 'hover effect'" and then undoing it. I found a fix with the information you provided for pausing the history ;).

Reply:
In the end, the problem is related to the fact that when we perform an "undo," the widgets are not restarted. So, we end up with a DOM that has changed, but certain JavaScript variables still hold their old values (as in this case, the SVGs saved in variables to avoid fetching them again for every mouseover).

cc @qsm-odoo

Reply:
To check if https://github.com/odoo/odoo/pull/145465 would solve the bug or not.

Reply:
> To check if #145465 would solve the bug or not.

No, that wouldn't solve it ... `widgets_start_request` and `widgets_stop_request` are only triggered when the mutation type is `remove` or `add`. So it won't be triggered on attribute or class changes and therefore won't solve my problem.

cc @Guillaume-gdi

Reply:
I add this to my todo list... When an attributes changes we should probably restart the public widgets :+1: